internal/client/gitlab.go

```diff
@@ -135,57 +157,73 @@ func (c *gitlabClient) CreateFile(
 	commitAuthor config.CommitAuthor,
 	repo Repo,
 	content []byte, // the content of the formula.rb
-	path, // the path to the formula.rb
+	fileName, // the path to the formula.rb
 	message string, // the commit msg
 ) error {
-	fileName := path
-
 	projectID := repo.Name
 	if repo.Owner != "" {
 		projectID = repo.Owner + "/" + projectID
 	}
 
-	// Use the project default branch if we can get it...otherwise, just use
-	// 'master'
-	var branch, ref string
+	log.
+		WithField("projectID", projectID).
+		Debug("project id")
+
+	var branch, defaultBranch string
+	var branchExists bool
 	var err error
 	// Use the branch if given one
 	if repo.Branch != "" {
 		branch = repo.Branch
+		branchExists, err = c.checkBranchExists(ctx, repo, branch)
+		if err != nil {
+			return err
+		}
+
+		// Retrieving default branch because we need it for `start_branch`
+		if !branchExists {
+			defaultBranch, err = c.getDefaultBranch(ctx, repo)
+			if err != nil {
+				return err
+			}
+		}
+
+		log.
+			WithField("projectID", projectID).
+			WithField("branch", branch).
+			WithField("branchExists", branchExists).
+			Debug("using given branch")
 	} else {
 		// Try to get the default branch from the Git provider
 		branch, err = c.getDefaultBranch(ctx, repo)
 		if err != nil {
-			// Fall back to 'master' 😭
-			log.
-				WithField("fileName", fileName).
-				WithField("projectID", projectID).
-				WithField("requestedBranch", branch).
-				WithError(err).
-				Warn("error checking for default branch, using master")
-			ref = "master"
-			branch = "master"
+			return err
 		}
-	}
-	ref = branch
-	opts := &gitlab.GetFileOptions{Ref: &ref}
-	castedContent := string(content)
 
-	log.
-		WithField("projectID", projectID).
-		WithField("ref", ref).
-		WithField("branch", branch).
-		Debug("projectID at brew")
+		defaultBranch = branch
+		branchExists = true
 
-	log.
-		WithField("projectID", projectID).
-		Info("pushing")
+		log.
+			WithField("projectID", projectID).
+			WithField("branch", branch).
+			Debug("no branch given, using default branch")
+	}
+
+	// If the branch doesn't exist, we need to check the default branch
+	// because that's what we use as `start_branch` later if the file needs
+	// to be created.
+	opts := &gitlab.GetFileOptions{Ref: &defaultBranch}
+	if branchExists {
+		opts.Ref = &branch
+	}
+	castedContent := string(content)
```

Comment:
probably better to name this stringContents, or something like that.

would also move it closer to where it's used

Reply:
Done! :) Though it was there before 🙃